docs/BEST_PRACTICES.md

```diff
@@ -2,11 +2,15 @@
 This is a collection of tips, advice, gotchas and other best practices for using the [Jenkins Pipeline plugin](https://github.com/jenkinsci/workflow-plugin/blob/master/README.md). Contributions and comments are happily accepted.
 
 # General tips and advice
-* Do everything that makes sense there within `stage`s. This will make your builds easier to visualize, debug, etc.
+* Do everything that makes sense there within `stage`s. This will make your builds easier to visualize, debug (e.g it isolates failures to a specific stage), display data (e.g it gets the execution time per build and per stage), etc.
```

Comment:
Stages do not intrinsically offer failure isolation.  In declarative they *may* but it's not guaranteed.

Reply:
done